src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/TypeSymbolExtensions.cs

```diff
@@ -300,5 +300,25 @@ public static (ImmutableArray<ITypeSymbol> TypeArguments, ImmutableArray<Nullabl
             }
             return (typeArguments.ToImmutable(), nullableAnnotations.ToImmutable());
         }
+
+        /// <summary>
+        /// Returns if the type is of the string representation of a type
+        /// </summary>
+        public static bool IsOfType(this INamedTypeSymbol type, string typeName)
+        {
+            if (typeName.Contains('<') || typeName.Contains('+') || typeName.Contains('/'))
+                throw new ArgumentException($"Cannot handle type name in the format provided: {typeName}");
```

Comment:
```suggestion
            if (typeName.Contains('<') || typeName.Contains('+') || typeName.Contains('/'))
                throw new ArgumentException("Cannot match generic or nested types", nameof(typeName));
```

Reply:
I'm not sure if that's the right message to give, because it can handle nested types and open generics as long as the delimiter is `.` instead of `+` or `/` and the generics use the `` `1`` notation instead of brackets.

Reply:
Yeah, you're right. Let's do the following instead (just including the argument name in the parameter list):

```suggestion
            if (typeName.Contains('<') || typeName.Contains('+') || typeName.Contains('/'))
                throw new ArgumentException($"Cannot handle type name in the format provided: {typeName}", nameof(typeName));
```